simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -85,6 +85,50 @@ HdMapUtils::HdMapUtils(
     lanelet::utils::query::shoulderLanelets(lanelet::utils::query::laneletLayer(lanelet_map_ptr_));
 }
 
+auto HdMapUtils::gelAllCanonicalizedLaneletPoses(
+  const traffic_simulator_msgs::msg::LaneletPose & lanelet_pose) const
+  -> std::vector<traffic_simulator_msgs::msg::LaneletPose>
+{
+  std::vector<traffic_simulator_msgs::msg::LaneletPose> canonicalized_all;
+  if (lanelet_pose.s < 0) {
+    if (const auto ids = getPreviousLaneletIds(lanelet_pose.lanelet_id); ids.empty()) {
+      return canonicalized_all;
+    } else {
+      for (const auto id : ids) {
+        const auto lanelet_pose_tmp = traffic_simulator::helper::constructLaneletPose(
+          id, lanelet_pose.s + getLaneletLength(id), 0);
```

Comment:
It seems, this line overwrites offset values and push_back in line 102.
I think offset values should be copy from lanelet_pose.

```suggestion
          id, lanelet_pose.s + getLaneletLength(id), lanelet_pose.offset);
```